### PR TITLE
Modify error rate's alert query that use UPI protocol

### PR DESCRIPTION
**Pull request by tiopramayudi**
Due to knative revision metrics treated all grpc response code as 200, we need to modify alert query regarding error rates. Modification is by using `istio_requests_total` metric if `grpc` protocol being used and using existing `revision_request_count` query if `http` protocol being  used

### CODE REVIEW

**Comment by krithika369** [comment]
> if grpc protocol being used and using existing revision_request_count query if http protocol being used

Since we are creating an Istio VS for the router from the Turing codebase, I think we can assume the Istio metric will always be there. Given this, shall we default to using the Istio metrics for both http and grpc instead of selectively handling this?

**Comment by tiopramayudi** [comment]
> > if grpc protocol being used and using existing revision_request_count query if http protocol being used
> 
> Since we are creating an Istio VS for the router from the Turing codebase, I think we can assume the Istio metric will always be there. Given this, shall we default to using the Istio metrics for both http and grpc instead of selectively handling this?

Update this using metric from `istio_requests_count` but still need to do the condition since we rely on different labels for status code between grpc and http

**Comment by krithika369** [comment]
Not important to do this, but for avoiding any confusing, can we update the sample alert in https://github.com/caraml-dev/turing/blob/main/api/turing/service/alert_service_test.go#L34, to no longer use `revision_request_count` ?

File: api/turing/models/alert.go [hunk 1]
```diff
@@ -146,20 +146,28 @@ func getAlertExpr(metric Metric, env string, rev string, threshold float64) stri
   service_name=~"%s-[0-9]*"
 }[1m])) by (le)) %s %f
 `, env, rev, getAlertOperator(metric), threshold)
-
 	case MetricErrorRate:
-		return fmt.Sprintf(`sum(rate(revision_request_count{
```
Comment (krithika369) [inline api/turing/models/alert.go:151]: What about the `revision_request_latencies_bucket` metric with UPI? Is it expected to still work, or does it have to be changed?

**Review by krithika369 — APPROVED** [review]
Left some comments. The rest LGTM. Thanks, @tiopramayudi !

**Comment by tiopramayudi** [comment]
> Not important to do this, but for avoiding any confusing, can we update the sample alert in https://github.com/caraml-dev/turing/blob/main/api/turing/service/alert_service_test.go#L34, to no longer use `revision_request_count` ?

The alert that must use `istio` metrics is only the error rate due to revision metric could not differentiate  ok or fail status code when using grpc, hence the only query that change only the error rate for throughput and latency we can still rely on the `revision` metrics

**Pull request merged**